### PR TITLE
feat: add offline node delete

### PR DESCRIPTION
**Pull request by niladrih**
Add DELETE /v0/nodes/{node_id} API for removing offline nodes and all
their resources (pools, replicas, nexuses, snapshots) without contacting
io-engine. Follows the pool purge pattern across all layers.

API:
  DELETE /v0/nodes/{id} with body {purge, accept, acceptVolumeLoss,
  acceptSnapshotLoss} returns NodeDeleteResult with aggregated volume
  and snapshot loss information.

CLI:
  kubectl mayastor delete node <id> --purge [--accept-data-loss]
  --accept-data-loss is shorthand for --accept-volume-loss
  --accept-snapshot-loss. The --yes prompt maps to accept=true.

Flow:
  1. Validate node is offline and cordoned
  2. Check purge/accept flags
  3. Mark node as Purging (crash-safe checkpoint)
  4. Auto-cordon each pool (replicas + snapshots) on behalf of the user
  5. Delegate to pool purge for each pool (replicas, snapshots, loss
     analysis all handled by existing pool purge code)
  6. On pool rejection: complete_destroy(Err) clears pending op but
     keeps Purging status for reconciler resume
  7. On success: complete_destroy removes node spec from store
  8. Remove node runtime wrapper

Crash recovery:
  - NodePurgeReconciler detects SpecStatus::Purging on startup, resumes
    with all accepts=true (validation was done before Purging was set)
  - PersistentStoreReconciler clears stale pending ops first
  - Individual pool crashes handled by existing pool purge reconciler

-------------------------

CLI:
  Add --show-impact to dry-run Node Purge

### CODE REVIEW

File: control-plane/plugin/src/resources/node.rs [hunk 1]
```diff
@@ -764,3 +764,84 @@ impl Label for Node {
...
+impl Delete for Node {
+    type ID = DeleteNodeArgs;
+
+    async fn del(id: &Self::ID, _ignore_not_found: bool, output: &OutputFormat) -> PluginResult {
+        // --accept-data-loss is shorthand for both volume and snapshot loss.
+        let accept_volume_loss = id.accept_volume_loss || id.accept_data_loss;
+        let accept_snapshot_loss = id.accept_snapshot_loss || id.accept_data_loss;
+        let body = openapi::models::DeleteNodeBody {
+            purge: Some(id.purge),
+            accept: Some(true),
```
Comment (Abhinandan-Purkait) [inline control-plane/plugin/src/resources/node.rs:816]: Why is this hardcoded to true? Is this not driven by any cli?

File: control-plane/agents/src/bin/core/controller/reconciler/node/purge.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        };
+        pool_guard
+            .cordon(context.registry(), cordon_request)
+            .await
+            .ok(); // Ignore if already cordoned.
+
+        let destroy_pool = DestroyPool::purge(pool.node(), pool.id().clone())
+            .with_accept(true)
+            .with_accept_volume_loss(true)
+            .with_accept_snapshot_loss(true);
```
Comment (Abhinandan-Purkait) [inline control-plane/agents/src/bin/core/controller/reconciler/node/purge.rs]: How do you know the original request validated this for all pools? If the original request went half way through all pools and then it crashed?

File: control-plane/agents/src/bin/core/node/service.rs [hunk 3]
```diff
@@ -450,4 +460,158 @@ impl Service {
...
+                }
+            }
+        }
+
+        // 7. All pools purged — mark node as Purging and complete deletion.
+        //    We mark Purging only after pools are gone so that a pre-flight
+        //    rejection (e.g. missing accept_volume_loss) doesn't leave the
+        //    node in a dirty Purging state.
+        //    In the resume case the node is already Purging — start_destroy_for_purge
+        //    re-logs the pending op so complete_destroy can commit.
```
Comment (Abhinandan-Purkait) [inline control-plane/agents/src/bin/core/node/service.rs]: Who will resume this if it fails above and we never marked the state purging until this point?

File: control-plane/agents/src/common/errors.rs [hunk 4]
```diff
@@ -489,6 +489,25 @@ pub enum SvcError {
...
+    #[snafu(display("Cannot delete node {node_id}: node must be cordoned first"))]
+    NodeNotCordoned { node_id: NodeId },
+
+    #[snafu(display(
+        "Cannot delete node {node_id}: node has resources. Use purge=true to delete all resources"
+    ))]
+    NodeHasResources { node_id: NodeId },
+
+    #[snafu(display(
+        "Cannot delete node {node_id}: node has {pool_count} pool(s), confirm=true required"
```
Comment (Abhinandan-Purkait) [inline control-plane/agents/src/common/errors.rs]: Is that accept=true or confirm=true?

Comment (Abhinandan-Purkait) [reply control-plane/agents/src/common/errors.rs]: Also, I did not see that option in the plugin

Comment (niladrih) [reply control-plane/agents/src/common/errors.rs]: That's a miss, it'll be accept=true.

The plugin delete flow now has a Y/n question (even for volume delete). If that question is answered with a Yes (or user uses the `--yes` option), we assume it's an accept=true for pools may have replicas and the user accepts it.

File: control-plane/plugin/src/resources/node.rs (hunk 1 above)
Comment (niladrih) [reply control-plane/plugin/src/resources/node.rs:816]: The plugin delete flow now has a Y/n question (even for volume delete). If that question is answered with a Yes (or user uses the `--yes` option), we assume it's an accept=true for pools may have replicas and the user accepts it. If the code has reached this stage, it's already passed the Y/n questions and the user has answered with a yes.

**Comment by tiagolobocastro** [comment]
I feel like I keep repeating myself but please add description to the commit and PR...

File: control-plane/agents/src/bin/core/node/service.rs (hunk 3 above)
Comment (niladrih) [reply control-plane/agents/src/bin/core/node/service.rs]: The pools are set to SpecStatus::Purging, so once the agent-core restarts, the pool reconciler should be able to finish the purge.

File: control-plane/agents/src/common/errors.rs (hunk 4 above)
Comment (niladrih) [reply control-plane/agents/src/common/errors.rs]: Made this change.

Comment (Abhinandan-Purkait) [reply control-plane/agents/src/common/errors.rs]: This errror message needs to be fixed right? accept=true

**Review by Abhinandan-Purkait — APPROVED** [review]
Looks mostly good to me, one thing I feel could be changed is error mapping, everything seems to result in CONFLICT.

File: control-plane/agents/src/common/errors.rs (hunk 4 above)
Comment (niladrih) [reply control-plane/agents/src/common/errors.rs]: Yes I have done that?

File: control-plane/agents/src/bin/core/controller/reconciler/node/purge.rs (hunk 2 above)
Comment (niladrih) [reply control-plane/agents/src/bin/core/controller/reconciler/node/purge.rs]: Pools are marked SpecStatus::Purged. The pool reconciler will pick up pending purges after we recover from the crash.

**Comment by niladrih** [comment]
This change has been added, and is no longer required.
